src/CombFilter/CombFilter.cpp

```diff
@@ -0,0 +1,93 @@
+//
+// Created by Rose Sun on 2/12/22.
+//
+
+#include "CombFilter.h"
+#include "RingBuffer.h"
+#include "ErrorDef.h"
+
+CCombFilterBase::CCombFilterBase (float fMaxDelayLengthInS, float fSampleRateInHz, int iNumChannels) :
+fMaxDelayLengthInS(fMaxDelayLengthInS),
+fSampleRateInHz(fSampleRateInHz),
+fDelayLengthInS(0.f),
+iNumChannels(iNumChannels)
+{
+    this->reset ();
+    bufferLength = static_cast<long>(round(fMaxDelayLengthInS * fSampleRateInHz));
+
+    ppRingBuffer = new CRingBuffer<float>*[iNumChannels];
+    for (int i=0; i<iNumChannels; ++i)
+        ppRingBuffer[i] = new CRingBuffer<float>(bufferLength);
+}
+
+Error_t CCombFilterBase::setParam (CCombFilterIf::FilterParam_t eParam, float fParamValue)
+{
+    switch (eParam) {
+        case CCombFilterIf::FilterParam_t::kParamDelay:
+            if (fParamValue > fMaxDelayLengthInS)
+                return Error_t::kFunctionInvalidArgsError;
+            fDelayLengthInS = fParamValue;
+            delayLength = static_cast<long>(round(fSampleRateInHz * fParamValue));
+            for (int i=0; i < iNumChannels; ++i)
+                ppRingBuffer[i]->setWriteIdx(1.F*delayLength + ppRingBuffer[i]->getReadIdx());
```

Comment:
Interesting choice of initialization. I personally would have gone with a method that fills the RingBuffer with 0s of `length = delayInSamples`. This makes it easier in the `process()` method to perform addition with the ring buffer elements.